collector/perf_linux.go

```diff
@@ -401,8 +457,9 @@ func (c *perfCollector) updateHardwareStats(ch chan<- prometheus.Metric) error {
 }
 
 func (c *perfCollector) updateSoftwareStats(ch chan<- prometheus.Metric) error {
-	for cpu, profiler := range c.perfSwProfilers {
-		cpuStr := fmt.Sprintf("%d", cpu)
+	for _, profiler := range c.perfSwProfilers {
+		cpuid := c.swProfilerCpuMap[&profiler]
```

Comment:
So, I'm new to go and this appears to be a bug.

`&profiler` here refers to the address of the `profiler` object itself and not the address of the `perf.SoftwareProfiler` that was inserted into `c.swProfilerCpuMap`.

This leads to the incorrect CPU id being retrieved. I'm not really sure how to deal with this as I know basically 0 golang (sorry). Would love any suggestions you folks have on the best way to fix this.